extended/src/main/java/io/kubernetes/client/extended/informer/PageSharedInformerFactory.java

```diff
@@ -0,0 +1,118 @@
+package io.kubernetes.client.extended.informer;
+
+import com.google.common.collect.Lists;
+import com.google.gson.reflect.TypeToken;
+import io.kubernetes.client.extended.pager.Pager;
+import io.kubernetes.client.extended.pager.Pager.PagerParams;
+import io.kubernetes.client.informer.ListerWatcher;
+import io.kubernetes.client.informer.SharedInformerFactory;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.util.CallGenerator;
+import io.kubernetes.client.util.CallGeneratorParams;
+import io.kubernetes.client.util.Watch;
+import io.kubernetes.client.util.exception.ObjectMetaReflectException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import okhttp3.Call;
+import okhttp3.OkHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** informer list request support pagination like client-go(default limit 500). */
+public class PageSharedInformerFactory extends SharedInformerFactory {
+
+  private static final Logger log = LoggerFactory.getLogger(PageSharedInformerFactory.class);
+
+  public PageSharedInformerFactory() {
+    super();
+  }
+
+  public PageSharedInformerFactory(ApiClient apiClient) {
+    super(apiClient);
+  }
+
+  public PageSharedInformerFactory(ApiClient client, ExecutorService threadPool) {
+    super(client, threadPool);
+  }
+
+  protected <ApiType, ApiListType> ListerWatcher<ApiType, ApiListType> listerWatcherFor(
+      final CallGenerator callGenerator,
+      final Class<ApiType> apiTypeClass,
+      final Class<ApiListType> apiListTypeClass) {
+
+    if (apiClient.getHttpClient().readTimeoutMillis() > 0) {
+      // set read timeout zero to ensure client doesn't time out
+      OkHttpClient httpClient =
+          apiClient.getHttpClient().newBuilder().readTimeout(0, TimeUnit.MILLISECONDS).build();
+      apiClient.setHttpClient(httpClient);
+    }
+
+    return new ListerWatcher<ApiType, ApiListType>() {
```

Comment:
This feels like a pretty big chunk of code to be an anonymous class, perhaps break it out as it's own class? It's pretty hard to test it as is.

Reply:
same with [SharedInformerFactory](https://github.com/kubernetes-client/java/blob/69ca6fd7cca4f8188ec9b490530e53cc9b0dea09/util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java#L131-L136)

Reply:
PageSharedInformerFactory  and PagerCallGeneratorParams use like client-go, add page default limit size (500)，this factory use for scale data。

I wanted to enhance the new features by changing the original code as little as possible.Without affecting the original function。

submit the basic implementation first, and then change other related implementations。